R/message_generators.R

```diff
@@ -3,59 +3,237 @@
 # incorrect/user code, which is always at hand). Cases:
 
-# missing argument
-missing_argument <- function(this_call, that, that_name = NULL) {
-  this_call <- prep(this_call)
-  that <- prep(that)
+# bad argument name
+bad_argument_name <- function(this_call, 
+                              this, 
+                              this_name,
+                              enclosing_call = NULL,
+                              enclosing_arg = NULL) { # only if the user supplied one (to match user code)
+  
+  # f(1, g(1, h(b = i(1))))
+  # f(1, a = g(1, a = h(ba = i(1)), bb = i(2)))
+  
+  # In f(1, g(1, h(b = i(1)))), h() accepts more than one argument that begins with b.
+  # As a result, R cannot figure out which argument you want to pass i(1) to.
+  # Check how you spelled b, or write out the full argument name.
+  
+  # {intro}{this_call} accepts more than one argument that begins with {this_name}.
+  # As a result, R cannot figure out which argument you want to pass {this} to.
+  # Check how you spelled {this_name}, or write out the full argument name.
 
-  if (!is.null(that_name) && that_name != "")
-    that <- paste(that_name, "=", that)
+  intro <- build_intro(.call = enclosing_call, .arg = enclosing_arg)
 
-  if (grepl("\\(\\)", that))
-    that <- paste("a call to", that)
+  this <- prep(this)
+  this_call <- prep(this_call)
+  
+  if (grepl("\\(\\)", this))
+    this <- paste("a call to", this)
+  
+  glue::glue_data(
+    list(
+      intro = intro,
+      this_call = this_call,
+      this_name = this_name,
+      this = this
+    ),
+    "{intro}{this_call} accepts more than one argument name that begins ",
+    "with {this_name}. As a result, R cannot figure out which ", 
+    "argument you want to pass {this} to. Check how you spelled ",
+    "{this_name}, or write out the full argument name."
+  )
+}
 
-  glue::glue("Your call to {this_call} should include {that} ",
-             "as one of its arguments. You may have ",
-             "referred to it in a different way, or left out an important ",
-             "argument name.")
+# WHAT TO DO IF THE MISSING ARGUMENT DOESN"T HAVE A NAME IN THE SOLUTION?
+# missing argument
+missing_argument <- function(this_call, 
+                             that_name = NULL,
+                             enclosing_call = NULL,
+                             enclosing_arg = NULL) {
+  
+  # f(1, g(1, h(i(1))))
+  # f(1, a = g(1, a = h(a = i(1)), b = i(2)))
+  
+  # "In g(1, h(i(1))), Your call to h() should include b",
+  # "as one of its arguments. You may have referred to it ",
+  # "in a different way, or left out an important argument name."
+  
+  # "{intro}Your call to {this_call} should include {that_name} ",
+  # "as one of its arguments. You may have referred to it ",
+  # "in a different way, or left out an important argument name."
+  
+  intro <- build_intro(.call = enclosing_call, .arg = enclosing_arg)
+  intro <- ifelse(intro == "", "Y", paste0(intro, "y"))
```

Comment:
Store this as a `your_char` variable and add it to the glue statement.  `"{intro}{your_char}our..."`